chapters/03-talking_to_servers/readme.md

```diff
@@ -0,0 +1,678 @@
+
+  <button type="submit">Solve!</button>
+</form>
+```
+
+And that's it! We made a form completely from scratch! If you are still
+interested as to why we need the `authenticity_token` input and why
+it has to be a specific value you can read into it by searching for
+Cross Site Request Forgery, or CSRF for short.
+
+Let's now go back to the controller and use the data that we sent via the form.
+To access data sent to the server we use the `params` object which is
+present in all Rails controllers. It looks and feels much like a regular hash
+but has additional features that we will cover in the next chapter. For
+now we can think of it and use it just like a hash.
+
+We could implement the solution finding algorithm in the following manner:
+
+```ruby
+def create
+  a = params[:a].to_i
+  b = params[:b].to_i
+  c = params[:c].to_i
+  d = params[:d].to_i
+
+  solution = solve(a, b, c, d)
+  solution = Marshal.dump(solution)
+  solution = Base64.encode64(solution)
+
+  redirect_to "/polynomials/#{solution}"
+end
+
+def solve(a, b, c, d)
+  require 'cmath'
+
+  a = a.to_r
+  b = b.to_r
+  c = (c - d).to_r
+
+  first = -1 * b
+  second = CMath.sqrt(b**2 - 4 * a * c) / (2 * a)
+
+  [first - second, first + second]
+end
+```
+
+There are a few tricky parts here. First is the `to_r` it converts any number
+to a rational number this is important because second degree polynomials have
+solutions with imaginary components. Then there is `CMath` it allows us to
+find imaginary roots of numbers, the regular `Math` root method throws an
+error if the input is negative. Finally there is `redirect_to` which you saw
+in the previous chapter. It redirects the user to the show page with the
+solution. But our solution is an array of two imaginary numbers! The
+browser can't understand that, therefor we use `Marshall` and `Base64`
+to convert it to a `String` the browser understands. `Marshall` converts Ruby
+objects to a compact binary-format or in layman's terms - to ones and zeroes.
+While `Base64` converts the ones and zeroes to letters.
+
+Now we have to do the same, but in reverse, in the show method to get the
+solutions out of the id.
+
+```ruby
+def show
+  @solution = params[:id]
+  @solution = Base64.decode64(@solution)
+  @solution = Marshal.restore(@solution)
+end
+```
+
+Notice that we used an instance variable. We learned in the last chapter that,
+using them, we can pass values from the controller into the view.
+
+Finally we can do the show view:
+
+```html
+<h1>The solutions!</h1>
+
+<h2>1. <%= @solution.first %></h2>
+<h2>2. <%= @solution.last %></h2>
+```
+
+![Polynomial submission](./images/new_polynomial_5.png)
+
+It only prints out the first and the second solution to the problem. Nothing
+to write home about. But with this we have a fully functioning polynomial
+solution finder written completely by hand! This was a learning experience, but
+we still have open questions! Why did Rails prefix the input's names with `post`
```

Comment:
Unclear how you got to this question as 'post' wasn't mentioned earlier except as a HTTP method.